addons/steps-dom/tests/integration/steps-test.ts

```diff
@@ -80,8 +78,274 @@ module('Integration | Component | steps', function (hooks) {
+        <textarea data-test-field></textarea>
+      `);
+
+      const tuple = findByLabel('Field');
+
+      await steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple);
+
+      assert.dom(tuple[0][0]).hasValue('foo');
+    });
+
+    test('Fill a contenteditable normally', async function (assert) {
+      await render(hbs`
+        <div contenteditable data-test-field></div>
+      `);
+
+      const tuple = findByLabel('Field');
+
+      await steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple);
+
+      assert.dom(tuple[0][0]).hasText('foo');
+    });
+
+    test('Fill an input inside given element', async function (assert) {
+      await render(hbs`
+        <div data-test-field>
+          <input>
+        </div>
+      `);
+
+      const tuple = findByLabel('Field');
+
+      await steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple);
+
+      assert.dom(find('input')).hasValue('foo');
+    });
+
+    test('Fill a textarea inside given element', async function (assert) {
+      await render(hbs`
+        <div data-test-field>
+          <textarea></textarea>
+        </div>
+      `);
+
+      const tuple = findByLabel('Field');
+
+      await steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple);
+
+      assert.dom(find('textarea')).hasValue('foo');
+    });
+
+    test('Fill a contenteditable inside given element', async function (assert) {
+      await render(hbs`
+        <div data-test-field>
+          <div contenteditable></div>
+        </div>
+      `);
+
+      const tuple = findByLabel('Field');
+
+      await steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple);
+
+      assert.dom(find('[contenteditable]')).hasText('foo');
+    });
+
+    test('Should throw on missing element', async function (assert) {
+      await render(hbs``);
+
+      const tuple = findByLabel('Field');
+
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      assert.rejects(
+        steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple),
+        /Expected `Field` to match 1 element\(s\), but 0 found/
+      );
+    });
+
+    test('Should throw when matching element is not editable and does not contain an editable', async function (assert) {
+      await render(hbs`<div data-test-field></div>`);
+
+      const tuple = findByLabel('Field');
+
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      assert.rejects(
+        steps['When I (?:fill|type) "(.*)" into $emberBddLabel']('foo', tuple),
+        /Expected element to be fillable or have exactly one fillable child, but 0 fillable children found/
       );
     });
   });
+
+  //
+
+  module('When I move the mouse pointer into $emberBddLabel', function () {
+    interface TestContextMouseEnter extends TestContext {
+      action?: SinonSpy;
+    }
+
+    test('Normal', async function (this: TestContextMouseEnter, assert) {
```

Comment:
@simonihmig This test in `addons/steps-dom` fails for the mouseenter step, can't figure out why.

Can you please have a look when you have time?

Reply:
The specific issue is that the action `{{on "mouseenter" this.action}}` won't fire.

Reply:
@lolmaus I don't see a reason why this shouldn't work. `mouseenter` is a special event in that it doesn't bubble up. That was an issue when using the event delegation pattern, i.e. registering a single event listener at a root element (e.g body) to catch all events below it. A while ago I did a PR to enable mouseenter events in Ember's EventDispatcher without jQuery, which was not working for this reason. The test coverage was involving, as I had to replicate what the events fired by a real browser would be, see https://github.com/emberjs/ember.js/pull/16999/files#diff-01b9a4fcff882974a0d09c7e5bd8a38a1729802457d197a8746b0b51374fb0b8. But I don't know if this is of any relevance here. With `{{on ...}}` we don't rely on the EventDispatcher and thus event delegation, so this shouldn't be an issue...

What puzzles me: CI is actually green! So does this test not run in CI yet? Or does it only fail locally in a real browser?

Reply:
This was a typo in `mouseeenter`. 😔